job/jobcache.go

```diff
@@ -1,36 +1,86 @@
 package job
 
 import (
+	"os"
+	"os/signal"
 	"sync"
+	"time"
 )
 
-var (
-	AllJobs = &JobCache{
-		Jobs:   make(map[string]*Job),
-		rwLock: sync.Mutex{},
+type JobCache interface {
+	Get(id string) *Job
```

Comment:
@ajvb It's impossible to fail to get a `*Job`?  What if one is requested that doesn't exist?  Probably want the signature to be `Get(id string) error`, and similar for most of the others.